### PR TITLE
Add casync tests and remove 'experimental' note from docs

### LINKED ISSUE #671: Is casync support still experimental?
Hi, this is not an issue but rather a question to the maintainers. 

I see that casync support was [added in v0.4](https://github.com/rauc/rauc/releases/tag/v0.4), and the documentation claims that [it is in experimental phase](https://rauc.readthedocs.io/en/latest/advanced.html#rauc-casync-support). Is it still the case with the most recent version, v1.4, after more than 2 years from the initial release?

### PR DESCRIPTION
**Pull request by ejoerns**
The docs were saying casync support is still experimental. While casync was always basically working, this disclaimer was mainly added due to missing test coverage for casync-related code.

Add the missing tests and update the documentation.

Based on #956 
Fixes #671

### CODE REVIEW

**Comment by jluebbe** [comment]
I've rebased it on master, as #956 is merged.

**Review by jluebbe — CHANGES_REQUESTED** [review]
I'll also push minor wording edits in the docs.

File: test/update_handler.c [hunk 1]
```diff
@@ -296,6 +389,18 @@ static void test_update_handler(UpdateHandlerFixture *fixture,
 		g_assert(test_make_filesystem(fixture->tmpdir, "image.ext4"));
 	} else if (g_strcmp0(test_pair->imagetype, "tar.bz2") == 0) {
 		g_assert_true(test_prepare_dummy_archive(fixture->tmpdir, "image.tar.bz2", "testfile.txt"));
+	} else if (g_strcmp0(test_pair->imagetype, "img.caibx") == 0) {
+		gchar* storepath = test_prepare_dummy_caibx(fixture->tmpdir, "image.img.caibx");
+		g_assert_nonnull(storepath);
+
+		r_context_conf()->install_info->mounted_bundle = g_new0(RaucBundle, 1);
+		r_context_conf()->install_info->mounted_bundle->storepath = storepath;
```
[the remark this answers is not in the document]
Comment (ejoerns) [reply test/update_handler.c:397]: I've added freeing of the bundle struct to the already-existing cleanup handling

File: test/update_handler.c [hunk 2]
```diff
@@ -924,6 +1042,35 @@ int main(int argc, char *argv[])
...
+			test_update_handler,
+			update_handler_fixture_tear_down);
+	/* casync directory index tests */
+	g_test_add("/update_handler/casync/caidx_to_ext4",
+			UpdateHandlerFixture,
+			&testpair_matrix[61],
+			update_handler_fixture_set_up,
+			test_update_handler,
+			update_handler_fixture_tear_down);
+	/* FIXME: only works with "--without=sec-time" set for "casync make"*/
```
[the remark this answers is not in the document]
Comment (ejoerns) [reply test/update_handler.c:1067]: Not fully unsupported but It seems that it requires at least adding `casync-args="--without=sec-time"` to the `rauc convert` call.
I have not investigated much deeper than to the casync `src/cadecoder.c` line

``` c
if (timespec_to_nsec(st.st_mtim) / granularity != read_le64(&child->entry->mtime) / granularity) 
        return -EOPNOTSUPP;
```

that is responsible for the quite imprecise casync error message

    Failed to run synchronizer: Operation not supported

**Review by jluebbe — COMMENTED** [review]
The bundle struct cleanup was added. Otherwise the changes were just a rebase.

**Pull request merged**